zapr.go

```diff
@@ -70,33 +70,66 @@ type zapLogger struct {
 	// NB: this looks very similar to zap.SugaredLogger, but
 	// deals with our desire to have multiple verbosity levels.
 	l *zap.Logger
+
+	// logNumericFieldName controls whether the numeric logr level is
+	// added to each log message and with which key.
+	logNumericFieldName string
+
+	// logErrorFieldName is the field name used for the error in
+	// Logger.Error calls.
+	logErrorFieldName string
```

Comment:
One thought: would it make sense to put the constant options into a struct and then just point to that struct in all copies of `zapLogger`?

I have no idea whether that is premature optimization and how it might influence  overhead in those functions which do a deep copy of `zapLogger` (`WithCallDepth`, `WithName`, etc.).

Reply:
copying small structs is VERY hard to measure.  If you think it might matter, craft a benchmark - you're counting nanoseconds :)

Reply:
I don't think it matters, but wanted to point it out just in case that I am wrong.